src/components/Camera/AICamera/AICamera.js

```diff
@@ -171,14 +171,22 @@ const AICamera = ( {
   const handleTakePhoto = useCallback( async ( ) => {
     await logStage( sentinelFileName, "take_photo_start" );
     setHasTakenPhoto( true );
-    setAICameraSuggestion( result );
+    // this feels a little duplicative, but we're currently using aICameraSuggestion
+    // to show the loading screen in Suggestions *without* setting an observation.taxon,
+    // and we're using visionResult to populate ObsEdit *with* the taxon
+    // before aICameraSuggestion has finished being stored.
+    // would be nice to refactor and set this more uniformly once the UX is more stable
+    // and we're fully certain we don't want to populate observation.taxon on Suggestions -> ObsEdit
+    setAICameraSuggestion( showPrediction && result );
     await takePhotoAndStoreUri( {
       replaceExisting: true,
       inactivateCallback: () => setInactive( true ),
-      navigateImmediately: true
+      navigateImmediately: true,
+      visionResult: showPrediction && result
```

Comment:
Since `result` can be `StoredResult | null` I would change here to `showPrediction ? result : null` in that way we don't send up a boolean (= false).